### PR TITLE
[editor] Remove obsolete `EditorDecorationsService`

### LINKED ISSUE #1785: Remove unused `EditorDecorationsService`
the `EditorDecorationsService` was introduced initially to apply editor decorations in a more convenient way when tracking changes of documents via the `Workspace` API. because `DidChangeTextDocumentParams` do not have any reference to the editor, the `EditorDecorationsService` will try to find an editor for the given uri. 

after resolving performance issues with the the computation of data for editor decorations in #1506, there is no client of `EditorDecorationsService` left. instead of using this indirection, it makes more sense to [track editors](https://github.com/theia-ide/theia/blob/165e6a7622d098634b7d557fb87b598a04eadf48/packages/merge-conflicts/src/browser/merge-conflicts-provider.ts#L42), and invoke `TextEditor.deltaDecorations` directly.

I think we should remove `EditorDecorationsService`. OTOH there is a `EditorDecorationStyle` utility in `editor-decorations-service.ts`, which is quite useful to create editor style dynamically. this should be considered to be moved.

the implementation of `TextEditor.deltaDecorations(DeltaDecorationParams)` in `MonacoDiffEditor` should be revised. `DeltaDecorationParams.uri` is only there to decide which editor should be decorated, which is redundant in case of `MonacoEditor`.

### PR DESCRIPTION
**Pull request by AlexTugarev**
Clean up. `TextEditor.deltaDecorations` should be used instead.

Fixes #1785

### CODE REVIEW

File: packages/editor/src/browser/editor-decorations.ts [hunk 1]
```diff
@@ -5,83 +5,9 @@
...
-}
-
-export enum TrackedRangeStickiness {
-    AlwaysGrowsWhenTypingAtEdges = 0,
-    NeverGrowsWhenTypingAtEdges = 1,
-    GrowsOnlyWhenTypingBefore = 2,
-    GrowsOnlyWhenTypingAfter = 3,
-}
-
 export interface EditorDecoration {
```
Comment (akosyakov) [inline packages/editor/src/browser/editor-decorations.ts]: The file name should after the main concept: `editor-decoration.ts`

File: packages/editor/src/browser/editor-decorations.ts [hunk 2]
```diff
@@ -157,6 +83,36 @@ export interface DecorationOverviewRulerOptions {
...
+}
+
+export enum TrackedRangeStickiness {
+    AlwaysGrowsWhenTypingAtEdges = 0,
+    NeverGrowsWhenTypingAtEdges = 1,
+    GrowsOnlyWhenTypingBefore = 2,
+    GrowsOnlyWhenTypingAfter = 3,
+}
+
+export abstract class EditorDecorator {
```
Comment (akosyakov) [inline packages/editor/src/browser/editor-decorations.ts]: Should not it be `@injectable` as well? Could you double check seems like inversify dropped this requirement for super classes.

Comment (akosyakov) [reply packages/editor/src/browser/editor-decorations.ts]: Maybe move to another file.

File: packages/editor/src/browser/editor.ts [hunk 3]
```diff
@@ -110,7 +110,7 @@ export interface SetDecorationParams {
 }
 
 export interface DeltaDecorationParams {
-    uri: string;
+    uri?: string;
```
Comment (akosyakov) [inline packages/editor/src/browser/editor.ts]: How about removing it?

Comment (AlexTugarev) [reply packages/editor/src/browser/editor.ts]: @akosyakov it depends on, how to deal with diff editor. should we just let the implementor do the work to get to the wanted editor part of the diff editor? then `deltaDecorations` of the diff editor would be no-op? Or should it log a warning, or even throw?

File: packages/editor/src/browser/editor-decorations.ts (hunk 1 above)
Comment (AlexTugarev) [reply packages/editor/src/browser/editor-decorations.ts]: I've updated and splitted up a bit more, see new sub-folder `decorations`.

File: packages/editor/src/browser/editor.ts (hunk 3 above)
Comment (AlexTugarev) [reply packages/editor/src/browser/editor.ts]: I removed it, and changed diff editor's `deltaDecorations` to be a no-op.

File: packages/editor/src/browser/decorations/editor-decorator.ts [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+ */
+
+import { injectable } from "inversify";
+import { TextEditor } from "../editor";
+import { EditorDecoration } from "./editor-decoration";
+
+@injectable()
+export abstract class EditorDecorator {
+
+    protected appliedDecorations = new Map<string, string[]>();
```
Comment (akosyakov) [inline packages/editor/src/browser/decorations/editor-decorator.ts]: `readonly`?

**Comment by AlexTugarev** [comment]
@jbicker I've updated and removed a parameter for `deltaDecorations` call in search and replace tree widget. Maybe you want to try out.

**Pull request merged**